tensordict/prototype/tensorclass.py

```diff
@@ -209,18 +215,26 @@ def wrapper(self, *args, batch_size, device=None, **kwargs):
     return wrapper
 
 
-def _from_tensordict_wrapper(expected_keys):
-    def wrapper(cls, tensordict):
+def _to_tensorclass_wrapper(expected_keys):
+    def wrapper(cls, tensordict, non_tensordict=None):
         if not all(key in expected_keys for key in tensordict.keys()):
             raise ValueError(
                 f"Keys from the tensordict ({set(tensordict.keys())}) must "
                 f"correspond to the class attributes ({expected_keys})."
             )
+        if non_tensordict is not None:
+            if not all(key in expected_keys for key in non_tensordict.keys()):
+                raise ValueError(
+                    f"Keys from the non-tensor data ({set(non_tensordict.keys())}) must "
+                    f"correspond to the class attributes ({expected_keys})."
+                )
         # bypass initialisation. this means we don't incur any overhead creating an
         # empty tensordict and writing values to it. we can skip this because we already
         # have a tensordict to use as the underlying tensordict
         tc = cls.__new__(cls)
         tc.__dict__["tensordict"] = tensordict
+        if non_tensordict is not None:
```

Comment:
is this if needed? shouldn't be initialised in all cases. That is 
`tc.__dict__["non_tensordict"] = non_tensordict if non_tensordict else {}`

Reply:
> > @vmoens I think the set attribute updates are specific to the object and hence if you change data_nest.v = "x" , data.y.v will still be "y"
> 
> are you sure? Since we are not copying by default it should reflect the changes. Anyway it is worth adding a test and document the behaviour
Please ignore the above comments. In the test case there was a clone happening. Yes the above behavior will hold